libgraph/include/katana/Properties.h

```diff
@@ -699,5 +699,126 @@ struct ArrayProperty {
+        array.length());
+  }
+
+  size_t size() const { return length_; }
+
+  // The functions below return (const) references to the beginning
+  // of a particular index's array (See ArrayRef).
+
+  reference GetValue(size_t index) {
+    return ArrayRef(ComputePointerLocation(index), array_size);
+  }
+
+  const_reference GetValue(size_t index) const {
+    return ArrayRef(ComputePointerLocation(index), array_size);
+  }
+
+  reference operator[](size_t index) {
+    return ArrayRef(ComputePointerLocation(index), array_size);
+  }
+
+  const_reference operator[](size_t index) const {
+    return ArrayRef(ComputePointerLocation(index), array_size);
+  }
+
+private:
+  FixedSizeBinaryPODArrayView(uint8_t* ptr, size_t length)
+      : ptr_(ptr), length_(length) {}
+
+  /// Pointer to the raw data contained by the FixedSizedBinaryArray.
+  uint8_t* ptr_;
+  /// Number of elements in the array
+  size_t length_;
+
+  /// Returns the correct pointer location for a particular index since
+  /// this class will know how many bytes each array element should take
+  PODType* ComputePointerLocation(size_t index) {
+    constexpr size_t element_byte_size = sizeof(PODType) * array_size;
+    return reinterpret_cast<PODType*>(ptr_ + (element_byte_size * index));
+  }
+};
+
+/// This is an property where each row has some fixed number of POD types with
+/// an underlying FixedSizedBinaryArray arrow representation. Constrasts with
+/// the ArrayProperty in that it does not use LargeLists (variable size is
+/// technically allowed there) and that the current implementation of ArrayProperty
+/// only supports doubles.
+///
+/// @tparam PODType POD type the array will contain
+/// @tparam array_size The fixed size of the property array.
+template <typename PODType, size_t array_size>
+struct FixedSizedBinaryPODArrayProperty
+    : public Property<
+          arrow::FixedSizeBinaryType,
+          FixedSizeBinaryPODArrayView<PODType, array_size>> {
+  /// Creates the table that contains a array_size elements of the PODType
+  /// for each row as a FixedSizeBinary.
+  static katana::Result<std::shared_ptr<arrow::Table>> Allocate(
+      size_t num_rows, const std::string& name) {
+    // creates the type of the necessary size
+    constexpr size_t binary_size = sizeof(PODType) * array_size;
+    auto fixed_size_type = KATANA_CHECKED_CONTEXT(
+        arrow::FixedSizeBinaryType::Make(binary_size),
+        "failed to make fixed size type of size {}", binary_size);
+
+    arrow::FixedSizeBinaryBuilder fixed_sized_binary_builder(fixed_size_type);
+    KATANA_CHECKED(fixed_sized_binary_builder.AppendEmptyValues(num_rows));
+
+    std::shared_ptr<arrow::Array> array_of_fixed_size_binaries =
+        KATANA_CHECKED_CONTEXT(
+            fixed_sized_binary_builder.Finish(),
+            "failed to finish fixed size binary builder");
+
+    return katana::Result<std::shared_ptr<arrow::Table>>(arrow::Table::Make(
```

Comment:
`return arrow::Table::Make(`
should work fine, there's no arrow::Result involved